sdk/core/core-amqp/src/auth/sas.ts

```diff
@@ -78,7 +84,98 @@ export class SharedKeyCredential {
    * @param {string} connectionString - The EventHub/ServiceBus connection string
    */
   static fromConnectionString(connectionString: string): SharedKeyCredential {
-    const parsed = parseConnectionString<ServiceBusConnectionStringModel>(connectionString);
-    return new SharedKeyCredential(parsed.SharedAccessKeyName, parsed.SharedAccessKey);
+    const parsed = parseConnectionString<
+      ServiceBusConnectionStringModel & { SharedAccessSignature: string }
+    >(connectionString);
+
+    if (parsed.SharedAccessSignature == null) {
+      return new SharedKeyCredential(parsed.SharedAccessKeyName, parsed.SharedAccessKey);
+    } else {
+      return new SharedAccessSignatureCredential(parsed.SharedAccessSignature);
+    }
+  }
+}
+
+/**
+ * A credential that takes a SharedAccessSignature of two forms:
+ * `SharedAccessSignature sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
+ * or
+ * `sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
+ *
+ * @internal
+ * @ignore
+ */
+export class SharedAccessSignatureCredential extends SharedKeyCredential {
+  private _accessToken: AccessToken;
+
+  /**
+   * @param sharedAccessSignature A shared access signature of the form `sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
+   */
+  constructor(sharedAccessSignature: string) {
+    super("", "");
+
+    const {
+      keyName,
+      expiry: expiresOnTimestamp,
+      token
+    } = SharedAccessSignatureCredential.extractSasProperties(sharedAccessSignature);
+
+    this.keyName = keyName;
+    this.key = "<none>";
+    this._accessToken = {
+      token,
+      expiresOnTimestamp
+    };
+
+    this.renewable = false;
+  }
+
+  /**
+   * Retrieve a valid token for authenticaton.
+   *
+   * @param _audience Not used for SAS tokens.
+   */
+  getToken(_audience: string): AccessToken {
+    return this._accessToken;
+  }
+
+  private static extractSasProperties(
+    sharedAccessSignature: string
+  ): {
+    expiry: number;
+    keyName: string;
+    token: string;
+  } {
+    let expiry: number | undefined;
+    let keyName: string | undefined;
+
+    sharedAccessSignature = sharedAccessSignature.replace(/^SharedAccessSignature\s+/, "");
+
+    for (const part of sharedAccessSignature.split("&")) {
+      const [key, value] = part.split("=");
+
+      if (expiry != null && keyName != null) {
+        break;
+      }
+
+      switch (key) {
+        case "se":
+          expiry = Number(decodeURIComponent(value));
+          break;
+        case "skn":
+          keyName = decodeURIComponent(value);
+          break;
```

Comment:
Are these the only two fields in SAS?
(Storage SAS has more.)

Reply:
These are the only two that we specifically need and even then it's mostly for two "required" fields that are exposed from SharedKeyCredential.

It's quite possible we could omit them.

Reply:
I removed this code altogether - we don't need to parse the underlying token now that I've made two changes:
1. Don't validate sharedKeyName in connection config (from core-amqp) when we pass a SAS.
2. Return 0 for expiry date when we don't want token renewal.